### PR TITLE
Maintenancev21.12/perfs

### PR DESCRIPTION
**Pull request by 6l20**
## PR description

- Add performance and stress tests to maintenance/v21.12

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -437,9 +479,9 @@ workflows:
           docker-password: ${CLOUDSMITH_PASS_RW}
           filters:
             branches:
-              only: main
+              only: maintenancev21.12/perfs
             tags:
-              only: /^v.*/
+              only: /^t.*/
```
Comment (ggarri) [inline .circleci/config.yml:484]: why this is now prefix by `t`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -437,9 +479,9 @@ workflows:
           docker-password: ${CLOUDSMITH_PASS_RW}
           filters:
             branches:
-              only: main
+              only: maintenancev21.12/perfs
```
Comment (ggarri) [inline .circleci/config.yml:482]: please undo

File: .circleci/config.yml [hunk 3]
```diff
@@ -490,10 +534,42 @@ workflows:
...
+            - tanuki-orchestrate-staging
+          requires:
+            - run-staging-e2e
+          filters:
+            tags:
+              only: /^t.*/
+            branches:
+              ignore: /.*/
+      - run-performance:
+          name: perf-over-staging
```
Comment (ggarri) [inline .circleci/config.yml:554]: avoid using shorcuts

File: Makefile [hunk 4]
```diff
@@ -297,10 +305,16 @@ nginx:
 down-nginx:
 	@docker-compose -f scripts/deps/docker-compose-tools.yml rm --force -s -v nginx nginx-prometheus-exporter
 
-vegeta:
+
+# Explicitly make API_URL to all shell instances
+export $API_URL := ${API_URL}
```
Comment (ggarri) [inline Makefile:310]: THis is not needed

File: Makefile [hunk 5]
```diff
@@ -297,10 +305,16 @@ nginx:
...
+
+vegeta: tools
+	@envsubst < scripts/vegeta/test > scripts/vegeta/test.target
 	@mkdir -p build/vegeta
-	@cat scripts/vegeta/test | vegeta attack -format=http -duration=30s -rate=200/s | tee build/vegeta/results.bin | vegeta report
+	@cat scripts/vegeta/test.target | vegeta attack -format=http -duration=30s -rate=200/s | tee build/vegeta/results.bin | vegeta report
```
Comment (ggarri) [inline Makefile:315]: I cannot see this new file anywhere

File: Makefile [hunk 6]
```diff
@@ -112,9 +117,9 @@ lint-tools: ## Install linting tools
 	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin v1.27.0
 
 tools: lint-tools ## Install test tools
-	@GO111MODULE=on go get github.com/golang/mock/mockgen@v1.4.3
-	@GO111MODULE=on go get github.com/swaggo/swag/cmd/swag@v1.7.0
-	@GO111MODULE=on go get github.com/tsenart/vegeta
+	go install github.com/golang/mock/mockgen@v1.4.3
+	go install github.com/swaggo/swag/cmd/swag@v1.7.8
+	GO111MODULE=off go get -u github.com/tsenart/vegeta
```
Comment (ggarri) [inline Makefile:122]: use binaries https://github.com/tsenart/vegeta/releases

File: Makefile (hunk 4 above)
Comment (6l20) [reply Makefile]: Yes

File: Makefile (hunk 5 above)
Comment (6l20) [reply Makefile:311]: It is created by the envsubst process

File: Makefile (hunk 6 above)
Comment (ggarri) [reply Makefile:122]: did you check if you go.mod file gets modify after running this?

Comment (6l20) [reply Makefile]: Files get changes if you let it on only.
But I have changed to binary

File: Makefile [hunk 7]
```diff
@@ -163,6 +172,9 @@ bootstrap-deps: bootstrap ## Wait for dependencies to be ready
 gobuild-e2e: ## Build Orchestrate e2e Docker image
 	@CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o ./build/bin/test ./tests/cmd
 
+gobuild-stress: ## Build Orchestrate stress binary
```
Comment (ggarri) [inline Makefile:175]: this is idem to `gobuild-e2e`

File: Makefile [hunk 8]
```diff
@@ -112,9 +119,11 @@ lint-tools: ## Install linting tools
 	@curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $$(go env GOPATH)/bin v1.27.0
 
 tools: lint-tools ## Install test tools
-	@GO111MODULE=on go get github.com/golang/mock/mockgen@v1.4.3
-	@GO111MODULE=on go get github.com/swaggo/swag/cmd/swag@v1.7.0
-	@GO111MODULE=on go get github.com/tsenart/vegeta
+	go install github.com/golang/mock/mockgen@v1.4.3
+	go install github.com/swaggo/swag/cmd/swag@v1.7.8
+	@curl -sSfL $(VEGETA_BIN_URL) -o vegeta.tar.gz
```
Comment (ggarri) [inline Makefile:124]: create a separate commad for installing vegeta

File: Makefile (hunk 5 above)
Comment (ggarri) [reply Makefile:315]: for simplicity you can do:
```
envsubst < scripts/vegeta/test | vegeta attack -format=http -duration=30s -rate=200/s | tee build/vegeta/results.bin | vegeta report
```

**Review by ggarri — APPROVED** [review]

**Pull request merged**